lib/elixir/lib/collectable.ex

```diff
@@ -24,23 +24,54 @@ defprotocol Collectable do
   `Enumerable` protocol. `into/1` can be seen as the opposite of
   `Enumerable.reduce/3`. If `Enumerable` is about taking values out,
   `Collectable.into/1` is about collecting those values into a structure.
+
+  ## Examples
+
+  To show how to manually use the `Collectable` protocol, let's play with its
+  implementation for `MapSet`.
+
+      iex> {initial_acc, collector_fun} = Collectable.into(MapSet.new())
+      iex> updated_acc = Enum.reduce([1, 2, 3], initial_acc, fn(elem, acc) ->
```

Comment:
That is my usual rule @ericmj, when there are multiple clauses on multiple lines then I don't use parens but when there's one clause and the arguments are just after `fn` then I use parens because I often find it very hard to read otherwise (for example, when the `fn` is not the only argument to a function, like in this case). I just forgot that in Elixir we don't use parens, but fixed that now :)